src/looprestoration.rs

```diff
@@ -1779,3 +1792,212 @@ unsafe fn wiener_filter_neon<BD: BitDepthLooprestorationArm>(
+    }
+
+    #[cfg(target_arch = "x86_64")]
+    {
+        if flags & DAV1D_X86_CPU_FLAG_AVX2 == 0 {
+            return;
+        }
+
+        c.wiener[0] = BD::dav1d_wiener_filter7_avx2;
+        c.wiener[1] = BD::dav1d_wiener_filter5_avx2;
+
+        if BD::BITDEPTH == 8 || bpc == 10 {
+            c.sgr[0] = BD::dav1d_sgr_filter_5x5_avx2;
+            c.sgr[1] = BD::dav1d_sgr_filter_3x3_avx2;
+            c.sgr[2] = BD::dav1d_sgr_filter_mix_avx2;
+        }
+
+        if flags & DAV1D_X86_CPU_FLAG_AVX512ICL == 0 {
+            return;
+        }
+
+        c.wiener[0] = BD::dav1d_wiener_filter7_avx512icl;
+        c.wiener[1] = BD::dav1d_wiener_filter5_avx512icl;
+
+        if BD::BITDEPTH == 8 || bpc == 10 {
+            c.sgr[0] = BD::dav1d_sgr_filter_5x5_avx512icl;
+            c.sgr[1] = BD::dav1d_sgr_filter_3x3_avx512icl;
+            c.sgr[2] = BD::dav1d_sgr_filter_mix_avx512icl;
+        }
+    }
+}
+
+#[cfg(all(feature = "asm", any(target_arch = "arm", target_arch = "aarch64")))]
+#[inline(always)]
+fn loop_restoration_dsp_init_arm<BD: BitDepthLoopRestorationAsm>(
+    c: &mut Dav1dLoopRestorationDSPContext,
+    mut bpc: libc::c_int,
+) {
+    use crate::src::arm::cpu::DAV1D_ARM_CPU_FLAG_NEON;
+    // TODO(randomPoison): Import temporarily needed until init fns are deduplicated.
+    use crate::src::looprestoration::*;
+
+    let flags = dav1d_get_cpu_flags();
+
+    if flags & DAV1D_ARM_CPU_FLAG_NEON == 0 {
+        return;
+    }
+
+    cfg_if! {
+        if #[cfg(target_arch = "aarch64")] {
+            c.wiener[0] = BD::dav1d_wiener_filter7_neon;
+            c.wiener[1] = BD::dav1d_wiener_filter5_neon;
+        } else {
+            c.wiener[0] = wiener_filter_neon_erased::<BD>;
+            c.wiener[1] = wiener_filter_neon_erased::<BD>;
+        }
+    }
+
+    if BD::BITDEPTH == 8 || bpc == 10 {
+        c.sgr[0] = sgr_filter_5x5_neon_erased::<BD>;
+        c.sgr[1] = sgr_filter_3x3_neon_erased::<BD>;
+        c.sgr[2] = sgr_filter_mix_neon_erased::<BD>;
+    }
+}
+
+#[cfg(not(feature = "asm"))]
+pub(crate) trait BitDepthLoopRestorationAsm: BitDepth {}
+
+#[cfg(not(feature = "asm"))]
+impl BitDepthLoopRestorationAsm for BitDepth8 {}
+
+#[cfg(not(feature = "asm"))]
+impl BitDepthLoopRestorationAsm for BitDepth16 {}
+
+#[cold]
+pub(crate) unsafe fn dav1d_loop_restoration_dsp_init<BD: BitDepthLoopRestorationAsm>(
+    c: &mut Dav1dLoopRestorationDSPContext,
+    _bpc: libc::c_int,
+) {
+    c.wiener[1] = wiener_c_erased::<BD>;
+    c.wiener[0] = c.wiener[1];
+    c.sgr[0] = sgr_5x5_c_erased::<BD>;
+    c.sgr[1] = sgr_3x3_c_erased::<BD>;
+    c.sgr[2] = sgr_mix_c_erased::<BD>;
+
+    #[cfg(feature = "asm")]
+    cfg_if! {
+        if #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] {
+            loop_restoration_dsp_init_x86::<BD>(c, _bpc);
+        } else if #[cfg(any(target_arch = "arm", target_arch = "aarch64"))] {
+            loop_restoration_dsp_init_arm::<BD>(c, _bpc);
+        }
+    }
```

Comment:
```suggestion
    #[cfg(feature = "asm")]
    cfg_if! {
        if #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] {
            loop_restoration_dsp_init_x86::<BD>(c, bpc);
        } else if #[cfg(any(target_arch = "arm", target_arch = "aarch64"))] {
            loop_restoration_dsp_init_arm::<BD>(c, bpc);
        }
    }
    
    let _ = bpc;
```

Reply:
```suggestion
    #[cfg(feature = "asm")]
    {
        #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
        {
            loop_restoration_dsp_init_x86::<BD>(c, bpc);
        }
        #[cfg(any(target_arch = "arm", target_arch = "aarch64"))]
        {
            loop_restoration_dsp_init_arm::<BD>(c, bpc);
        }
    }
    
    let _ = bpc;
```

Superceding suggestion that incorporates the first one, too.  I'd prefer not to use `cfg_if!` since a bunch of things like formatting don't work inside of it.  It'd still be pretty clean without the `cfg_if!` so I don't think we need it here.